sycl/test-e2e/Basic/buffer/reinterpret.cpp

```diff
@@ -101,12 +101,19 @@ int main() {
     }
   }
 
-  try {
-    sycl::buffer<float, 1> buf_fl(r1d);
-    auto buf_d = buf_1d.reinterpret<double>(r2d);
-  } catch (sycl::invalid_object_error e) {
-    std::cout << "Expected exception has been caught: " << e.what()
-              << std::endl;
+  {
+    bool gotException = false;
+    try {
+      sycl::buffer<float, 1> buf_fl(r1d);
+      auto buf_d = buf_1d.reinterpret<double>(r2d);
+    } catch (sycl::exception e) {
+      if (e.code() == sycl::errc::invalid) {
+        std::cout << "Expected exception has been caught: " << e.what()
+                  << std::endl;
+        gotException = true;
+      }
+    }
+    assert(gotException);
```

Comment:
Nit; Currently, we do not know the difference between an unexpected exception being thrown and no exception being thrown. Following would address this.
```suggestion
      assert(e.code() == sycl::errc::invalid);
      std::cout << "Expected exception has been caught: " << e.what()
                << std::endl;
      gotException = true;
    }
    assert(gotException);
```